packages/ibm-products-web-components/src/components/interstitial-screen/interstitial-screen.ts

```diff
@@ -0,0 +1,216 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { LitElement, html, nothing } from 'lit';
+import { property, query, state } from 'lit/decorators.js';
+import { prefix } from '../../globals/settings';
+import '@carbon/web-components/es/components/modal/index.js';
+import HostListenerMixin from '@carbon/web-components/es/globals/mixins/host-listener.js';
+import { carbonElement as customElement } from '@carbon/web-components/es/globals/decorators/carbon-element.js';
+
+import styles from './interstitial-screen.scss?lit';
+import { SignalWatcher } from '@lit-labs/signals';
+import {
+  interstitialDetailsSignal,
+  resetInterstitialDetailsSignal,
+  updateInterstitialDetailsSignal,
+} from './interstitial-screen-context';
+
+export const blockClass = `${prefix}--interstitial-screen`;
+
+export type disableButtonConfigType = {
+  skip?: boolean;
+  back?: boolean;
+  next?: boolean;
+  start?: boolean;
+};
+/**
+ * interstitial-screen main component
+ * @element c4p-interstitial-screen
+ * @fires c4p-interstitial-opened -  The custom event triggered after loading the component.
+ * Its event.detail will provide you with carousal api methods for step navigation and method to disable any action button
+ * * @fires c4p-interstitial-beingclosed - The name of the custom event fired before interstitial is being closed upon a user gesture.
+ * Cancellation of this event stops the user-initiated action of closing the interstitial.
+ * @fires c4p-interstitial-closed - The name of the custom event fired after this tearsheet is closed upon a user gesture.
+
+ */
+
+@customElement(`${prefix}-interstitial-screen`)
+class CDSInterstitialScreen extends SignalWatcher(
+  HostListenerMixin(LitElement)
+) {
+  /**
+   * Specifies whether the component is shown as a full-screen
+   * experience, else it is shown as a modal by default.
+   */
+
+  @property({ type: Boolean, reflect: true, attribute: 'fullscreen' })
+  isFullScreen: boolean = false;
+  /**
+   * Specifies whether the component is currently open.
+   */
+  @property({ type: Boolean, reflect: true })
+  open: boolean = false;
+
+  @state()
+  stepDetails: Array<{ stepTitle: string; name?: string }> = [];
```

Comment:
Can we define a separate type?

```suggestion
interface StepDetail { stepTitle: string; name?: string }

  stepDetails: Array<StepDetail> = [];
```